subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4202,13 +4202,15 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 int lwm2m_parse_peerinfo(char *url, struct sockaddr *addr, bool *use_dtls)
 {
 	struct http_parser_url parser;
-#if defined(CONFIG_DNS_RESOLVER)
+#if defined(CONFIG_DNS_RESOLVER) || defined(CONFIG_NET_SOCKETS_OFFLOAD)
```

Comment:
This does not look right, the DNS resolving is now "done" even if is not enabled in config.

Reply:
Thanks for the comment. I added to define CONFIG_DNS_RESOLVER in my application and confirmed no issue using either hostname or IP address for LwM2M server. Thus I updated my commit with a new push.